dump/plugins/vlan.py

```diff
@@ -0,0 +1,123 @@
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) != 0:
+            for mem in ret["keys"]:
+                self.ret_temp[req.db]["keys"].append(mem)
+        else:
+            self.ret_temp[req.db]["tables_not_found"].append(req.table)
+        
+    def init_state_vlan_info(self, vlan_name):
+        req = MatchRequest(db="STATE_DB", table="VLAN_TABLE", key_pattern=vlan_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) != 0:
+            self.ret_temp[req.db]["keys"] = ret["keys"]
+        else:
+            self.ret_temp[req.db]["tables_not_found"] = [req.table]
+    
+    def init_state_vlan_member_info(self, vlan_name):
+        req = MatchRequest(db="STATE_DB", table="VLAN_MEMBER_TABLE", key_pattern=vlan_name+"*", ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) != 0:
+            for mem in ret["keys"]:
+                self.ret_temp[req.db]["keys"].append(mem)
+        else:
+            self.ret_temp[req.db]["tables_not_found"].append(req.table)
+
+    def init_asic_vlan_info(self, vlan_name):
+        # Convert 'Vlanxxx' to 'xxx'
+        if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+            self.ret_temp["ASIC_DB"]["tables_not_found"] =["ASIC_STATE:SAI_OBJECT_TYPE_VLAN"]
+            return
+        vlan_num = int(vlan_name[4:])
+        
+        # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", field="SAI_VLAN_ATTR_VLAN_ID", 
+                           value=str(vlan_num), ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) != 0:
+            self.ret_temp[req.db]["keys"] = ret["keys"]
+        else:
+            self.ret_temp[req.db]["tables_not_found"] = [req.table]
+    
+    def init_asic_vlan_member_info(self, vlan_name):
+        # Convert 'Vlanxxx' to 'xxx'
+        if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+            self.ret_temp["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER")
+            return
+        vlan_num = int(vlan_name[4:])
+        
+        # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", 
+                           field="SAI_VLAN_ATTR_VLAN_ID", value=str(vlan_num), ns=self.ns)
+        ret = self.match_engine.fetch(req)
+
+        # Find all the member tables whose vlan is the oid:0x... part of the table name just found 
+        if not ret["error"] and len(ret["keys"]) != 0:
+            req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER", key_pattern="*", 
```

Comment:
Can you please get SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID in return fields and dump the bridge port?

Reply:
Done.